brian2/codegen/runtime/cython_rt/templates/spikegenerator.pyx

```diff
@@ -1,39 +1,45 @@
 {% extends 'common.pyx' %}
 
-{% block template_support_code %}
-cdef int _start_idx = 0
-cdef double _current_time = 0.0
-{% endblock %}
-
 {% block maincode %}
 
-    {# USES_VARIABLES {_spikespace, N, t, dt, neuron_index, spike_time } #}
-     
-    global _start_idx, _current_time
-    
-    # Check if we have been reset, if so reset the cache
-    if t<_current_time:
-        _start_idx = 0
-    
-    _current_time = t
+    {# USES_VARIABLES {_spikespace, N, t, dt, neuron_index, spike_time, period, _lastindex } #}
 
     # TODO: We don't deal with more than one spike per neuron yet

```

Comment:
It turns out numpy's `bincount` is pretty straightforward and efficient for this task:
```Python
bincounts = np.bincount(timebins);bincounts = bincounts[bincounts!=0]
```
I did some tests with "timebin" arrays of size 1000000, and it took never more than ~7ms, and it seems to scale linearly for bigger arrays. I'll go ahead and implement something based on this approach.